geefcc/plots.py

```diff
@@ -0,0 +1,203 @@
+    """
+
+    # Figure
+    fig = plt.figure(figsize=figsize, dpi=dpi)
+    ax = plt.axes(projection=ccrs.PlateCarree())
+
+    # Extent
+    xmin, xmax, ymin, ymax = raster.x.min(), raster.x.max(), raster.y.min(), raster.y.max()
+    xbuffer = (xmax - xmin) * 0.025
+    ybuffer = (ymax - ymin) * 0.025
+    extent = [xmin-xbuffer, xmax+xbuffer, ymin-ybuffer, ymax+ybuffer]
+    ax.set_extent(extent, ccrs.PlateCarree())
+
+    # Plot raster
+    raster.plot(ax=ax, cmap=color_map, add_colorbar=False, transform=ccrs.PlateCarree(), **kwargs)
+
+    # Plot lat/lon grid
+    if grid:
+        gl = ax.gridlines(crs=ccrs.PlateCarree(), draw_labels=True,
+                      linewidth=0.1, color='k', alpha=1,
+                      linestyle='--')
+        gl.top_labels = False
+        gl.right_labels = False
+        gl.xformatter = LONGITUDE_FORMATTER
+        gl.yformatter = LATITUDE_FORMATTER
+        gl.xlabel_style = {'size': 8}
+        gl.ylabel_style = {'size': 8}
+
+    # Add borders
+    if borders:
+        ax.add_feature(BORDERS)
+
+    ax.add_feature(OCEAN, zorder=0)
+    ax.coastlines(linewidth=1)
+
+    plt.legend(handles=patches, bbox_to_anchor=(1.05, 1), loc=2, borderaxespad=0.)
+
+    return fig
+
+
+
+def _set_color_map_and_legend(colors: list[tuple], labels: dict):
+    """Set color map for raster plot as well as patches for legend.
+
+    This function sets the color map and patches that are used for plot.
+
+    :param colors: List of colors.
+    :param labels: Dictionary of labels associated with the colormap
+
+    :return: Color map.
+    :return: List of patches for legend.
+
+    """
+
+    # Colors
+    list_colors = [(1, 1, 1, 0)]  # transparent white for 0
+    cmax = 255.0  # float for division
+    for col in colors:
+        col_class = tuple([i / cmax for i in col])
+        list_colors.append(col_class)
+    color_map = ListedColormap(list_colors)
+    patches = [mpatches.Patch(facecolor=col, edgecolor="black",
+                              label=labels[i]) for (i, col) in enumerate(list_colors)]
+    return color_map, patches
+
+
+def plot_fcc(input_fcc_raster : str,
+    output_file : str = "fcc.png",
+    source : str = "tmf",
+    years  : list = [2000, 2010, 2020],
+    borders : bool = True,
+    zoom : tuple | None = None,
+    grid : bool = True,
+    colors : list[tuple] = [(255, 165, 0, 255), (227, 26, 28, 255), (34, 139, 34, 255)],
+    figsize : tuple = (11.69, 8.27),
+    dpi : int = 300,
+    **kwargs
+):
+    """Plot forest-cover change (fcc) map.
+
+    This function plots the forest-cover change map with 2
+    deforestation time-periods (2000 -> 2010 -> 2020 for example) plus
+    the remaining forest (3 classes).
+
+
+    :param input_fcc_raster: Path to fcc raster.
+    :param output_file: Name of the plot file.
+    :param source: Source of the forest cover data (tmf or gfc).
+    :param maxpixels: Maximum number of pixels to plot.
+    :param years: List of years for the fcc in the raster.
```

Comment:
Same as for the `source` argument, the `years` argument should be read from the `.tif` file itself. However, although the time stamp is present in the metadata of the tiles `.tif` files, they are not in the `.vrt` or reconstructed `.tif` files.

```
❯ gdalinfo out_tmf/forest_tiles/forest_1.tif
Driver: GTiff/GeoTIFF
Files: out_tmf/forest_tiles/forest_1.tif
Size is 1855, 1855
Coordinate System is:
...
Band 1 Block=256x256 Type=Int8, ColorInterp=Gray
  Metadata:
    time=2000-01-01T00:00:00.000000000
    data_var=forest_cover
Band 2 Block=256x256 Type=Int8, ColorInterp=Undefined
  Metadata:
    time=2010-01-01T00:00:00.000000000
    data_var=forest_cover
Band 3 Block=256x256 Type=Int8, ColorInterp=Undefined
  Metadata:
    time=2020-01-01T00:00:00.000000000
    data_var=forest_cover


❯ gdalinfo out_tmf/forest_tmf.tif
Driver: GTiff/GeoTIFF
Files: out_tmf/forest_tmf.tif
Size is 2305, 1923
Coordinate System is:
...
Band 1 Block=2305x1 Type=Int8, ColorInterp=Gray
Band 2 Block=2305x1 Type=Int8, ColorInterp=Undefined
Band 3 Block=2305x1 Type=Int8, ColorInterp=Undefined
```